File: InJava/app/src/main/java/com/barryzeha/pomodoroapp/model/repository/LocalDataSourceImpl.java
package com.barryzeha.pomodoroapp.model.repository;

import androidx.lifecycle.MutableLiveData;

import com.barryzeha.pomodoroapp.MyApp;
import com.barryzeha.pomodoroapp.model.TaskModel;
import com.barryzeha.pomodoroapp.model.database.TaskDAO;

import java.util.List;

/****
 * Project PomodoroApp
 * Created by Barry Zea H. on 18/10/2022
 * Copyright (c)  All rights reserved.
 ***/
public class LocalDataSourceImpl implements LocalDataSource{
    private final TaskDAO db= MyApp.taskDb.taskDAO();
    private final MutableLiveData<List<TaskModel>> taskList= new MutableLiveData<>();

    @Override
    public void saveTask(TaskModel taskModel) {
        db.saveTask(taskModel);
    }

    @Override
    public void callTasks() {
        taskList.postValue(db.getAllTask());
    }

    @Override
    public MutableLiveData<List<TaskModel>> getAllTask() {
        return taskList;
    }

    @Override
    public TaskModel getTask(int id) {
        return db.getTask(id);
    }

    @Override
    public void deleteAll() {
        db.clearHystoryTask();
    }

    @Override
    public void deleteTask(int id) {
        db.deleteTask(id);
    }
}

File: InKotlin/app/src/main/java/com/barryzeha/pomodoroapp/model/repository/LocalDataSourceImpl.kt
package com.barryzeha.pomodoroapp.model.repository

import androidx.lifecycle.MutableLiveData
import com.barryzeha.pomodoroapp.MyApp
import com.barryzeha.pomodoroapp.model.TaskModel
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.withContext

/****
 * Project PomodoroApp
 * Created by Barry Zea H. on 04/10/2022
 * Copyright (c)  All rights reserved.
 ***/
class LocalDataSourceImpl: LocalDataSource {
    private val db=MyApp.taskDB!!.taskDao()
    private val taskList : MutableLiveData<List<TaskModel>> = MutableLiveData()
    override suspend fun saveTask(taskModel: TaskModel)= withContext(Dispatchers.IO) {
        db.saveTask(taskModel)
    }

    override suspend fun getAllTask(): MutableLiveData<List<TaskModel>> = withContext(Dispatchers.IO) {
        taskList.postValue(db.getAllTask())
        return@withContext taskList
    }

    override suspend fun getTask(id: Int): TaskModel = withContext(Dispatchers.IO){
        return@withContext db.getTask(id)
    }

    override suspend fun deleteAll()= withContext(Dispatchers.IO){
       db.clearHistory()
    }

    override suspend fun deleteTask(id: Int) = withContext(Dispatchers.IO) {
        db.deleteTask(id)
    }
}